frontend/src/container/AdminUserSpacesConfig.jsx

```diff
@@ -71,38 +59,20 @@ export const AdminUserSpacesConfig = (props) => {
       if (space.memberList.find(u => u.id === props.userToEditId)) memberSpaces.push(space)
       else availableSpaces.push(space)
     })
-    setAvailableSpaceList(availableSpaces)
-    setDisplayedAvailableSpaceList(filterSpaceList(availableSpaces, availableSpaceListFilter))
-    setMemberSpaceList(memberSpaces)
-    setDisplayedMemberSpaceList(filterSpaceListWithUserRole(memberSpaces, memberSpaceListFilter))
-  }, [spaceList])
 
-  useEffect(() => {
-    setDisplayedAvailableSpaceList(filterSpaceList(availableSpaceList, availableSpaceListFilter))
-  }, [availableSpaceListFilter])
-
-  useEffect(() => {
-    setDisplayedMemberSpaceList(filterSpaceListWithUserRole(memberSpaceList, memberSpaceListFilter))
-  }, [memberSpaceListFilter])
+    const newSpaceList = []
 
-  const filterSpaceListWithUserRole = (list, filterList) => {
-    return list.filter(space => {
+    spaceList.forEach(space => {
       const member = space.memberList.find(u => u.id === props.userToEditId)
-      const userRole = ROLE_LIST.find(type => type.slug === member.role) || { label: '' }
-
-      const includesFilter = stringIncludes(filterList)
-
-      const hasFilterMatchOnUserRole = userRole && includesFilter(props.t(userRole.label))
-      const hasFilterMatchOnSpaceLabel = includesFilter(space.label)
-      const hasFilterMatchOnSpaceId = space.id && includesFilter(space.id.toString())
-
-      return (
-        hasFilterMatchOnUserRole ||
-        hasFilterMatchOnSpaceLabel ||
-        hasFilterMatchOnSpaceId
-      )
+      if (space.memberList.length > 0 && member) {
```

Comment:
I think you can just verify if `member` exists here:
If `space.memberList.length == 0` then you won't have any member

Reply:
This is not my code, and I think member is checked because even if `space.memberList.length` is not empty it is theoretically possible that member does not exist.

I don't really know since it's not my code :/